contracts/src/spaces/facets/membership/join/MembershipJoin.sol

```diff
@@ -109,10 +110,26 @@ abstract contract MembershipJoin is
     }
   }
 
+  function _getRequiredAmount() internal view returns (uint256) {
+    // Check if there are any prepaid memberships available
+    uint256 prepaidSupply = _getPrepaidSupply();
+    if (prepaidSupply > 0) return 0; // If prepaid memberships exist, no payment is required
+
+    // Get the current membership price based on total supply
+    uint256 price = _getMembershipPrice(_totalSupply());
+    if (price == 0) return 0; // If the price is zero, no payment is required
+
+    // Calculate the protocol fee
+    uint256 fee = _getProtocolFee(price);
+
+    // Return the higher of the price or fee to ensure at least the protocol fee is covered
+    return price > fee ? price : fee;
+  }
+
   function _validatePayment() internal view {
     if (msg.value > 0) {
```

Comment:
If `msg.value == 0`, it would revert inside `_chargeForJoinSpaceWithReferral` with `Membership__InsufficientPayment` instead of here.

Reply:
Yeah, since this is a cross chain call, it needs to revert at the node post side